pkg/patterns/declarative/pkg/applier/applylib.go

```diff
@@ -29,9 +38,11 @@ func (a *ApplySetApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 		switch arg {
 		case "--force":
 			opt.Force = true
-
+		case "--prune":
+			opt.Prune = true
 		default:
-			return fmt.Errorf("extraArg %q is not supported by the ApplySetApplier", arg)
+			// TODO(yuwenma): This skips the "--selector addons.configdelivery.anthos.io/MyCR=MyCRName"
+			klog.Errorf("extraArg %q is not supported by the ApplySetApplier", arg)
```

Comment:
I think you should specifically handle `--selector` and ignore it (perhaps with a warning), and keep the error on anything we don't know about.